Executing listQuarkusDependencies on Opening IDE happens and turns out to failure

```
13:15:48: Executing 'listQuarkusDependencies -c C:\Users\asus\repo\rir\rir-fias\15950601097609595862.gradle -q --console plain'...

FAILURE: Build failed with an exception.

* What went wrong:
The specified settings file 'C:\Users\asus\repo\rir\rir-fias\15950601097609595862.gradle' does not exist.
```

This happening to me for every Quarkus project for now 
What is this for?

Thank You!

--- a/gradle.properties
+++ b/gradle.properties
@@ -16,7 +16,7 @@ platformVersion=2023.3
 platformBundledPlugins=com.intellij.java, org.jetbrains.idea.maven, org.jetbrains.plugins.gradle, com.intellij.properties, org.jetbrains.plugins.yaml
 platformPlugins=com.redhat.devtools.intellij.telemetry:1.2.1.62
 # Gradle Releases -> https://github.com/gradle/gradle/releases
-gradleVersion=8.6
+gradleVersion=8.7
 channel=nightly
 quarkusVersion=3.15.1
 lsp4mpVersion=0.13.0

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

--- a/gradlew.bat
+++ b/gradlew.bat
@@ -43,11 +43,11 @@ set JAVA_EXE=java.exe
 %JAVA_EXE% -version >NUL 2>&1
 if %ERRORLEVEL% equ 0 goto execute
 
-echo.
-echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
-echo.
-echo Please set the JAVA_HOME variable in your environment to match the
-echo location of your Java installation.
+echo. 1>&2
+echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH. 1>&2
+echo. 1>&2
+echo Please set the JAVA_HOME variable in your environment to match the 1>&2
+echo location of your Java installation. 1>&2
 
 goto fail
 
@@ -57,11 +57,11 @@ set JAVA_EXE=%JAVA_HOME%/bin/java.exe
 
 if exist "%JAVA_EXE%" goto execute
 
-echo.
-echo ERROR: JAVA_HOME is set to an invalid directory: %JAVA_HOME%
-echo.
-echo Please set the JAVA_HOME variable in your environment to match the
-echo location of your Java installation.
+echo. 1>&2
+echo ERROR: JAVA_HOME is set to an invalid directory: %JAVA_HOME% 1>&2
+echo. 1>&2
+echo Please set the JAVA_HOME variable in your environment to match the 1>&2
+echo location of your Java installation. 1>&2
 
 goto fail
 

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java
@@ -90,7 +90,7 @@ public class QuarkusDeploymentSupport implements ClasspathResourceChangedManager
      * @param module            the module.
      * @param progressIndicator the progress indicator.
      */
-    public static void updateClasspathWithQuarkusDeployment(Module module, ProgressIndicator progressIndicator) {
+    public void updateClasspathWithQuarkusDeployment(Module module, ProgressIndicator progressIndicator) {
         if (module.isDisposed())
             return;
         LOGGER.info("Ensuring library to " + module.getName());
@@ -141,7 +141,7 @@ public class QuarkusDeploymentSupport implements ClasspathResourceChangedManager
     }
 
     @NotNull
-    private synchronized static CompletableFuture<Void> internalUpdateClasspathWithQuarkusDeployment(Module module) {
+    private synchronized CompletableFuture<Void> internalUpdateClasspathWithQuarkusDeployment(Module module) {
         CompletableFuture<Void> deploymentSupport = module.getUserData(QUARKUS_DEPLOYMENT_SUPPORT_KEY);
         if (!isOutOfDated(deploymentSupport)) {
             return deploymentSupport;

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/buildtool/gradle/AbstractGradleToolDelegate.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/buildtool/gradle/AbstractGradleToolDelegate.java
@@ -18,8 +18,10 @@ import com.intellij.execution.executors.DefaultRunExecutor;
 import com.intellij.ide.util.newProjectWizard.AddModuleWizard;
 import com.intellij.openapi.application.ReadAction;
 import com.intellij.openapi.externalSystem.model.execution.ExternalSystemTaskExecutionSettings;
+import com.intellij.openapi.externalSystem.service.execution.ProgressExecutionMode;
 import com.intellij.openapi.externalSystem.service.project.ProjectDataManager;
 import com.intellij.openapi.externalSystem.service.project.manage.ProjectDataImportListener;
+import com.intellij.openapi.externalSystem.task.TaskCallback;
 import com.intellij.openapi.externalSystem.util.ExternalSystemApiUtil;
 import com.intellij.openapi.externalSystem.util.ExternalSystemUtil;
 import com.intellij.openapi.module.ModifiableModuleModel;
@@ -116,14 +118,34 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
         Path outputPath = Files.createTempFile(null, ".txt");
         Path customBuildFile = generateCustomGradleBuild(getModuleDirPath(module), outputPath, deploymentIds);
         Path customSettingsFile = generateCustomGradleSettings(getModuleDirPath(module), customBuildFile);
+        TaskCallback callback = new TaskCallback() {
+            @Override
+            public void onSuccess() {
+                cleanCustomFiles();
+            }
+
+            @Override
+            public void onFailure() {
+                LOGGER.error("Failed to run custom gradle build");
+                cleanCustomFiles();
+            }
+
+            private void cleanCustomFiles() {
+                try {
+                    Files.delete(customBuildFile);
+                    Files.delete(customSettingsFile);
+                } catch (IOException e) {
+                    LOGGER.warn(e.getLocalizedMessage(), e);
+                }
+            }
+
+        };
         try {
-            collectDependencies(module, customBuildFile, customSettingsFile, outputPath, result);
+            collectDependencies(module, customSettingsFile, outputPath, result, callback);
         } catch (IOException e) {
             LOGGER.warn(e.getLocalizedMessage(), e);
         } finally {
             Files.delete(outputPath);
-            Files.delete(customBuildFile);
-            Files.delete(customSettingsFile);
         }
     }
 
@@ -149,37 +171,32 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
     /**
      * Collect all deployment JARs and dependencies through a Gradle specific task.
      *
-     * @param module          the module to analyze
-     * @param customBuildFile the custom Gradle build file with the specific task
-     * @param outputPath      the file where the result of the specific task is stored
-     * @param result          the list where to place results to
+     * @param module     the module to analyze
+     * @param outputPath the file where the result of the specific task is stored
+     * @param result     the list where to place results to
+     * @param callback   the callback to call after running the task
      * @throws IOException if an error occurs running Gradle
      */
-    private void collectDependencies(Module module, Path customBuildFile, Path customSettingsFile, Path outputPath, List<VirtualFile>[] result) throws IOException {
+    private void collectDependencies(Module module, Path customSettingsFile, Path outputPath, List<VirtualFile>[] result, TaskCallback callback) throws IOException {
         try {
             ExternalSystemTaskExecutionSettings executionSettings = new ExternalSystemTaskExecutionSettings();
             executionSettings.setExternalSystemIdString(GradleConstants.SYSTEM_ID.toString());
             executionSettings.setTaskNames(Collections.singletonList("listQuarkusDependencies"));
             executionSettings.setScriptParameters(String.format("-c %s -q --console plain", customSettingsFile.toString()));
             executionSettings.setExternalProjectPath(getModuleDirPath(module));
-            ExternalSystemUtil.runTask(
-                    executionSettings,
-                    DefaultRunExecutor.EXECUTOR_ID,
-                    module.getProject(),
-                    GradleConstants.SYSTEM_ID
-            );
+            ExternalSystemUtil.runTask(executionSettings,DefaultRunExecutor.EXECUTOR_ID,
+                module.getProject(),
+                GradleConstants.SYSTEM_ID, callback, ProgressExecutionMode.IN_BACKGROUND_ASYNC, false);
             try (BufferedReader reader = Files.newBufferedReader(outputPath)) {
                 String id;
 
                 while ((id = reader.readLine()) != null) {
                     String file = reader.readLine();
                     String[] ids = id.split(":");
-                    if (!isDependency(ModuleRootManager.getInstance(module), ids[0], ids[1])) {
-                        if (file != null) {
-                            VirtualFile jarFile = getJarFile(file);
-                            if (jarFile != null) {
-                                result[file.endsWith("sources.jar") ? SOURCES : BINARY].add(jarFile);
-                            }
+                    if (!isDependency(ModuleRootManager.getInstance(module), ids[0], ids[1]) && file != null) {
+                        VirtualFile jarFile = getJarFile(file);
+                        if (jarFile != null) {
+                            result[file.endsWith("sources.jar") ? SOURCES : BINARY].add(jarFile);
                         }
                     }
                 }
@@ -233,6 +250,7 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
         try (Reader reader = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
             content = IOUtils.toString(reader);
         } catch (IOException e) {
+            LOGGER.warn(e.getLocalizedMessage(), e);
         }
         content += System.lineSeparator() + "rootProject.buildFileName =\"" + customBuildFile.getFileName().toFile().getName() + "\"";
         Path customPath = Files.createTempFile(base, null, getScriptExtension());
@@ -252,7 +270,7 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
      * @return the content of the custom build file
      */
     private String appendQuarkusResolution(String content, Path outputPath, Set<String> deploymentIds) {
-        StringBuffer buffer = new StringBuffer(content);
+        StringBuilder buffer = new StringBuilder(content);
         buffer.append(System.lineSeparator());
         buffer.append(createQuarkusConfiguration()).append(System.lineSeparator());
         buffer.append("dependencies {").append(System.lineSeparator());
@@ -281,29 +299,27 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
      * @return the File object
      */
     public static File getDeploymentFile(String artifactId) {
-        String path = toPath(artifactId, File.separatorChar);
+        String path = toPath(artifactId);
         if (path != null) {
             return new File(M2_REPO, path);
         }
         return null;
     }
 
-    private static String toPath(String artifactId, char separator) {
+    private static String toPath(String artifactId) {
         String[] comps = artifactId.split(":");
         if (comps.length == 3) {
-            StringBuffer buffer = new StringBuffer(comps[0].replace('.', separator));
-            buffer.append(separator).append(comps[1]).append(separator).append(comps[2]).append(separator).append(comps[1]).append('-').append(comps[2]).append(".jar");
-            return buffer.toString();
+            return comps[0].replace('.', File.separatorChar) + File.separatorChar + comps[1] + File.separatorChar + comps[2] + File.separatorChar + comps[1] + '-' + comps[2] + ".jar";
         }
         return null;
     }
 
     private boolean isDependency(ModuleRootManager manager, String deploymentIdStr) {
-        return Stream.of(manager.getOrderEntries()).filter(entry -> entry instanceof LibraryOrderEntry && (GRADLE_LIBRARY_PREFIX + deploymentIdStr).equals(((LibraryOrderEntry) entry).getLibraryName())).findFirst().isPresent();
+        return Stream.of(manager.getOrderEntries()).anyMatch(entry -> entry instanceof LibraryOrderEntry && (GRADLE_LIBRARY_PREFIX + deploymentIdStr).equals(((LibraryOrderEntry) entry).getLibraryName()));
     }
 
     private boolean isDependency(ModuleRootManager manager, String groupId, String artifactId) {
-        return Stream.of(manager.getOrderEntries()).filter(entry -> entry instanceof LibraryOrderEntry && ((LibraryOrderEntry) entry).getLibraryName().startsWith(GRADLE_LIBRARY_PREFIX + groupId + ':' + artifactId)).findFirst().isPresent();
+        return Stream.of(manager.getOrderEntries()).anyMatch(entry -> entry instanceof LibraryOrderEntry && ((LibraryOrderEntry) entry).getLibraryName().startsWith(GRADLE_LIBRARY_PREFIX + groupId + ':' + artifactId));
     }
 
     @Override
@@ -381,16 +397,16 @@ public abstract class AbstractGradleToolDelegate implements BuildToolDelegate {
                 }
 
                 ReadAction.nonBlocking(() -> {
-                            List<Module> modules = new ArrayList<>();
-                            Module[] existingModules = ModuleManager.getInstance(project).getModules();
-                            for (Module module : existingModules) {
-                                // Check if the module is a Gradle project
-                                if (GradleRunnerUtil.isGradleModule(module) && isValidGradleModule(module)) {
-                                    modules.add(module);
-                                }
+                        List<Module> modules = new ArrayList<>();
+                        Module[] existingModules = ModuleManager.getInstance(project).getModules();
+                        for (Module module : existingModules) {
+                            // Check if the module is a Gradle project
+                            if (GradleRunnerUtil.isGradleModule(module) && isValidGradleModule(module)) {
+                                modules.add(module);
                             }
-                            listener.importFinished(modules);
-                        })
+                        }
+                        listener.importFinished(modules);
+                    })
                         .inSmartMode(project)
                         .submit(NonUrgentExecutor.getInstance());
             }

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/buildtool/maven/MavenToolDelegate.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/buildtool/maven/MavenToolDelegate.java
@@ -39,12 +39,13 @@ import org.jetbrains.idea.maven.project.MavenImportListener;
 import org.jetbrains.idea.maven.project.MavenProject;
 import org.jetbrains.idea.maven.project.MavenProjectsManager;
 import org.jetbrains.idea.maven.server.MavenEmbedderWrapper;
-import org.jetbrains.idea.maven.server.MavenServerManager;
 import org.jetbrains.idea.maven.utils.MavenUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.*;
+
+import static org.jetbrains.idea.maven.project.MavenEmbeddersManager.FOR_DEPENDENCIES_RESOLVE;
 
 public class MavenToolDelegate implements BuildToolDelegate {
     private static final Logger LOGGER = LoggerFactory.getLogger(MavenToolDelegate.class);
@@ -154,7 +155,7 @@ public class MavenToolDelegate implements BuildToolDelegate {
     private Set<MavenArtifact> resolveDeploymentArtifacts(Module module, MavenProject mavenProject, Set<MavenId> deploymentIds, String classifier, ProgressIndicator progressIndicator) {
         Set<MavenArtifact> deploymentArtifacts = new HashSet<>();
         try {
-            MavenEmbedderWrapper serverWrapper = MavenServerManager.getInstance().createEmbedder(module.getProject(), true, mavenProject.getDirectory());
+            MavenEmbedderWrapper serverWrapper = MavenProjectsManager.getInstance(module.getProject()).getEmbeddersManager().getEmbedder(mavenProject, FOR_DEPENDENCIES_RESOLVE);
             if (classifier != null) {
                 for (MavenId id : deploymentIds) {
                     deploymentArtifacts.add(serverWrapper.resolve(new MavenArtifactInfo(id, "jar", classifier), mavenProject.getRemoteRepositories()));

--- a/src/test/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/GradlePropertiesManagerLocationUsingVertxTest.java
+++ b/src/test/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/GradlePropertiesManagerLocationUsingVertxTest.java
@@ -37,7 +37,7 @@ public class GradlePropertiesManagerLocationUsingVertxTest extends GradleTestCas
     }
 
     @Test
-    public void testUsingVertxTest() throws Exception {
+    public void testUsingVertxTest() {
         Module javaProject = getModule("using-vertx.main");
         QuarkusDeploymentSupport.getInstance(javaProject.getProject()).updateClasspathWithQuarkusDeployment(javaProject, new EmptyProgressIndicator());
 

--- a/src/test/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/LSP4MPMavenModuleImportingTestCase.java
+++ b/src/test/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/LSP4MPMavenModuleImportingTestCase.java
@@ -31,7 +31,7 @@ public abstract class LSP4MPMavenModuleImportingTestCase extends MavenModuleImpo
     protected Module loadMavenProject(String projectName, boolean collectAndAddQuarkusDeploymentDependencies) throws Exception {
         Module module = createMavenModule(new File("projects/lsp4mp/projects/maven/" + projectName));
         if(collectAndAddQuarkusDeploymentDependencies) {
-            QuarkusDeploymentSupport.updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
+            QuarkusDeploymentSupport.getInstance(myTestFixture.getProject()).updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
         }
         return module;
     }

--- a/src/test/java/com/redhat/devtools/intellij/quarkus/completion/MavenApplicationPropertiesCompletionTest.java
+++ b/src/test/java/com/redhat/devtools/intellij/quarkus/completion/MavenApplicationPropertiesCompletionTest.java
@@ -81,7 +81,7 @@ public class MavenApplicationPropertiesCompletionTest extends MavenEditorTest {
 	protected Module loadMavenProject(String projectName, boolean collectAndAddQuarkusDeploymentDependencies) throws Exception {
 		Module module = createMavenModule(new File("projects/quarkus/projects/maven/" + projectName));
 		if(collectAndAddQuarkusDeploymentDependencies) {
-			QuarkusDeploymentSupport.updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
+			QuarkusDeploymentSupport.getInstance(myTestFixture.getProject()).updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
 		}
 		return module;
 	}

--- a/src/test/java/com/redhat/microprofile/psi/quarkus/QuarkusMavenModuleImportingTestCase.java
+++ b/src/test/java/com/redhat/microprofile/psi/quarkus/QuarkusMavenModuleImportingTestCase.java
@@ -31,7 +31,7 @@ public abstract class QuarkusMavenModuleImportingTestCase extends MavenModuleImp
     protected Module loadMavenProject(String projectName, boolean collectAndAddQuarkusDeploymentDependencies) throws Exception {
         Module module = createMavenModule(new File("projects/quarkus/projects/maven/" + projectName));
         if(collectAndAddQuarkusDeploymentDependencies) {
-            QuarkusDeploymentSupport.updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
+            QuarkusDeploymentSupport.getInstance(myTestFixture.getProject()).updateClasspathWithQuarkusDeployment(module, new EmptyProgressIndicator());
         }
         return module;
     }

